### PR TITLE
Update jest config per breaking changes

### PR DESCRIPTION
**Pull request by rekmarks**
We recently bumped `eslint-plugin-jest` to `24.x`, but didn't account for [breaking changes](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v24.0.0). This PR updates our `jest` config per the [`eslint-plugin-jest` readme](https://github.com/jest-community/eslint-plugin-jest/blob/master/README.md).

Summary of changes:
- Extends `recommended` and `style` rulesets, removes all rules that are enabled by them
  - Except such rules that we configure differently
- Move some deprecated rules into `jest/no-restricted-matchers` config

### CODE REVIEW

File: config/jest.js [hunk 1]
```diff
@@ -11,20 +11,23 @@ module.exports = {
...
     'jest/no-hooks': 'off',
     'jest/no-identical-title': 'error',
+    'jest/no-interpolation-in-shapshots': 'error',
     'jest/no-if': 'error',
     'jest/no-jasmine-globals': 'error',
     'jest/no-jest-import': 'error',
     'jest/no-large-snapshots': 'off',
     'jest/no-mocks-import': 'error',
     'jest/no-standalone-expect': 'error',
-    'jest/no-test-callback': 'error',
```
Comment (rekmarks) [inline config/jest.js:27]: Renamed to `no-done-callback`.

**Comment by brad-decker** [comment]
I believe the breaking changes regarding recommendations are only if you're using the recommended settings? If our goal is to match the recommendations, should we simply implement the recommended settings?
```
{
  "extends": ["plugin:jest/recommended"]
}
```

**Comment by rekmarks** [comment]
I think we decided to explicitly list every individual rule, but I'm not sure. @Gudahtt may have more context.

We do need to delete the `no-test-callback` rule, in any event.

**Comment by brad-decker** [comment]
On one hand, we'd be exposing ourselves to every update to recommendations Jest makes... on the other hand... we don't have to make every manual change to recommendations Jest makes if we intend to follow them. Sorry, @rekmarks, I'm not against this landing as is just want to understand before picking up work for #91 :D ... I really think we should just extend from their recommendations that are optimized for Prettier vs listing everything manually.

**Comment by rekmarks** [comment]
No worries @brad-decker! I agree with you. Let's discuss this async / next standup.

**Comment by brad-decker** [comment]
Ah, its unclear to me whether plugins override what you manually set, too. so these still may be inoperable.

**Comment by rekmarks** [comment]
@brad-decker per the docs, the specified rules override the extended ones: https://eslint.org/docs/user-guide/configuring#extending-configuration-files

**Comment by brad-decker** [comment]
So by specifying all core rules we make ourselves ineligible to take advantage of recommendations from plugins?

**Comment by brad-decker** [comment]
FWIW: we have a ton of rules that are making prettier do double work, I have been tinkering with it locally but nothing concrete to put up in PR yet (extension only). not related to this PR at all, and I'm gonna approve momentarily cause this should be escalated into a different thread.

**Review by brad-decker — APPROVED** [review]

**Pull request merged**